File: rust/src/models/usage_pricing.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::usage_pricing_model::UsagePricingModel;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct UsagePricing {
    pub model: UsagePricingModel,
}

impl UsagePricing {
    pub fn new(model: UsagePricingModel) -> Self {
        Self { model }
    }
}

File: typescript/src/models/usagePricing.ts
// this file is @generated
import { type UsagePricingModel, UsagePricingModelSerializer } from "./usagePricingModel";

export interface UsagePricing {
  model: UsagePricingModel;
}

export const UsagePricingSerializer = {
  _fromJsonObject(object: any): UsagePricing {
    return {
      model: UsagePricingModelSerializer._fromJsonObject(object["model"]),
    };
  },

  _toJsonObject(self: UsagePricing): any {
    return {
      model: UsagePricingModelSerializer._toJsonObject(self.model),
    };
  },
};
